src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random randomGenerator = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (randomGenerator.nextInt(5)) {
+            case 0:
+                return new Square(COLOR_SUPPLIER.getColorSupplier(),
+                        randomGenerator.nextDouble() * 10 + 1);
+            case 1:
+                return new Rectangle(COLOR_SUPPLIER.getColorSupplier(),
+                        randomGenerator.nextDouble() * 10 + 1,
+                        randomGenerator.nextDouble() * 10 + 1);
+            case 2:
+                return new RightTriangle(COLOR_SUPPLIER.getColorSupplier(),
+                        randomGenerator.nextDouble() * 10 + 1,
+                        randomGenerator.nextDouble() * 10 * 1);
+            case 3:
+                return new Circle(COLOR_SUPPLIER.getColorSupplier(),
+                        randomGenerator.nextDouble() * 10 + 1);

```

Comment:
Avoid passing expressions directly into constructors. Instead, assign the expression results to variables and pass those variables to the constructor. This improves code readability.